create SSOT for setup and install requires common

zancas opened the pull request:
Problem: Some pacakages are required both for setup
and install.  At least in the cases covered thus far
a single version should be available for both processes.

Analysis: Added "common_requires" list to store common
requirements.

Tests: Manual pip install and freeze

NOTE:  Also made flake8 and pep257 pass for seutp.py

zancas commented:
The second commit also fixed the originaton point of the Singleton (meta?) class.

dangle commented on rcli/autodetect.py:
```diff
@@ -70,9 +70,9 @@ def egg_info_writer(cmd, basename, filename):
     for k, v in six.iteritems(config):
         if v.lower() in ('y', 'yes', 'true'):
-            config[k] = True
+            config[k] = True  # type: ignore
```
What version of `mypy` are you using that makes this necessary? It would be better to change this to `# type: bool` though.

zancas replied:
```BASH
1515687676 || mammut mherder master[ ]
/home/mherder/rcli RC: 127 $
source .tox/py36/bin/activate
1515687689 || mammut mherder master[ ]
/home/mherder/rcli: 0 $
mypy --version
mypy 0.560
```

zancas replied:
Changing to type bool...

dangle replied:
Is that checked in?

zancas replied:
No it turns out it doesn't work.

dangle commented on rcli/config.py:
```diff
@@ -104,6 +104,7 @@ def distribution(self):
         """The distribution containing the currently active entry point."""
         if self.entry_point:
             return self.entry_point.dist
+        return None
```
Why the explicit return?

zancas replied:
pylint...   or flake8..  pylint I think.

dangle replied:
Are you running this with `tox`? That will make sure that you're running those with the correct settings.

zancas replied:
I am running tox, yes.

dangle commented on rcli/display.py:
```diff
@@ -106,7 +106,7 @@ def print_status(msg, color):
     except (KeyboardInterrupt, EOFError):
         raise
-    except:
+    except Exception:
```
Do we actually care about them being subclasses of `Exception`? We already have the two major `BaseException` subclasses above; does this cut anything out?

zancas replied:
I don't care, but pylint does.   Is it better to `except Exception:` or to `except:  # Ignore such-and-such`?

dangle replied:
You generally still need the broad-except ignore for `except Exception`.

zancas replied:
My local tests don't indicate this.

dangle commented on setup.py:
```diff
@@ -32,22 +34,17 @@
     packages=find_packages(exclude=['tests', 'docs']),
     install_requires=[
-        'typingplus >= 1.0.2, < 2',
+        'typet >= 0.3.4, < 4',
```
Update this to 0.3.5.

dangle commented:
To fix the py35 build, you'll need to add/change a couple of lines in `rcli/dispatcher.py`.

```python
import typing
```

should become

```python
import typingplus as typing
...
typing.upgrade_typing()
```

zancas commented:
OK adding py35 fix...

zancas commented:
OK @dangle I've addressed all your review comments (I think).   If you agree then please do the thingie that unblocks...   or else ping me with more comments.

zancas commented:
YAY!   All checks have passed, @dangle ...!

dangle commented on rcli/dispatcher.py:
```diff
@@ -11,7 +11,8 @@
 import types  # noqa: F401 pylint: disable=unused-import
-import typing  # noqa: F401 pylint: disable=unused-import
+
+import typingplus as typing  # noqa: F401 pylint: disable=unused-import
```
Move this down to be grouped with the external non-builtins. After colorama, or some such.

dangle commented:
You can change `import typingplus as typing  # noqa: F401 pylint: disable=unused-import` to just `import typingplus as typing` since it's being used now.

zancas commented:
OK

zancas commented:
Done.
